tests/Translator/Extractor/AdminExtractorTest.php

```diff
@@ -0,0 +1,123 @@
+use Sonata\AdminBundle\Translator\Extractor\AdminExtractor;
+use Symfony\Component\DependencyInjection\Container;
+use Symfony\Component\Translation\MessageCatalogue;
+
+final class AdminExtractorTest extends TestCase
+{
+    /**
+     * @var AdminExtractor
+     */
+    private $adminExtractor;
+
+    /**
+     * @var Pool
+     */
+    private $pool;
+
+    /**
+     * @var AdminInterface
+     */
+    private $fooAdmin;
+
+    /**
+     * @var AdminInterface
+     */
+    private $barAdmin;
+
+    /**
+     * @var BreadcrumbsBuilderInterface
+     */
+    private $breadcrumbsBuilder;
+
+    public function setUp(): void
+    {
+        $this->fooAdmin = $this->createMock(AdminInterface::class);
+        $this->barAdmin = $this->createMock(AdminInterface::class);
+
+        $container = new Container();
+        $container->set('foo_admin', $this->fooAdmin);
+        $container->set('bar_admin', $this->barAdmin);
+
+        $this->pool = new Pool($container, 'title', 'logo_title');
+        $this->pool->setAdminServiceIds(['foo_admin', 'bar_admin']);
+        $this->pool->setAdminGroups(['group' => [
+            'label_catalogue' => 'admin_domain',
+        ]]);
+
+        $this->breadcrumbsBuilder = $this->createMock(BreadcrumbsBuilderInterface::class);
+        $this->adminExtractor = new AdminExtractor($this->pool, $this->breadcrumbsBuilder);
+    }
+
+    public function testExtractEmpty(): void
+    {
+        $catalogue = new MessageCatalogue('en');
+
+        $this->adminExtractor->extract([], $catalogue);
+        $this->assertFalse($catalogue->has('foo', 'foo_admin_domain'));
+    }
+
+    public function testExtract(): void
+    {
+        $this->fooAdmin
+            ->method('getLabel')
+            ->willReturn('foo_label');
+        $this->fooAdmin
+            ->method('getTranslationDomain')
+            ->willReturn('foo_admin_domain');
+
+        $catalogue = new MessageCatalogue('en');
+
+        $this->adminExtractor->extract([], $catalogue);
+
+        $this->assertCount(2, $catalogue->getDomains());
+        $message = $catalogue->get('foo', 'foo_admin_domain');
+        $this->assertSame('foo', $message);
+
+        $this->assertTrue($catalogue->has('group', 'admin_domain'));
+        $this->assertTrue($catalogue->has('foo_label', 'foo_admin_domain'));
+    }
+
+    public function testExtractWithException(): void
+    {
+        $this->expectException(\RuntimeException::class);
+        $this->expectExceptionMessage('Foo throws exception');
+
+        $this->fooAdmin
+            ->method('getShow')
+            ->willReturnCallback(static function (): void {
+                throw new \RuntimeException('Foo throws exception');
+            });
+
+        $catalogue = new MessageCatalogue('en');
+
+        $this->adminExtractor->extract([], $catalogue);
+    }
+
+    public function testExtractCallsBreadcrumbs(): void
+    {
+        $this->breadcrumbsBuilder->expects($this->exactly(2 * 6))
```

Comment:
Any reason to use the math operation `2 * 6` instead its result?

Reply:
I have no idea TBH, the test was a copy of the other one, I'll check.